libcaf_io/test/automatic_connection.cpp

```diff
@@ -65,223 +75,282 @@ using pong_atom = atom_constant<atom("pong")>;
 
 */
 
-/*
-std::thread run_prog(const char* arg, uint16_t port, bool use_asio) {
-  return detail::run_sub_unit_test(invalid_actor,
-                                   test::engine::path(),
-                                   test::engine::max_runtime(),
-                                   CAF_XSTR(CAF_SUITE),
-                                   use_asio,
-                                   {"--port=" + std::to_string(port), arg});
-}
+namespace {
+
+constexpr uint16_t port_earth   = 12340;
+constexpr uint16_t port_mars    = 12341;
+constexpr uint16_t port_jupiter = 12342;
 
-// we run the same code on all three nodes, a simple ping-pong client
-struct testee_state {
-  std::set<actor> buddies;
-  uint16_t port = 0;
-  const char* name = "testee";
+// Used for the tests with the test backend.
+class config : public actor_system_config {
+public:
+  config() {
+    load<caf::io::middleman, io::network::test_multiplexer>();
+    set("scheduler.policy", caf::atom("testing"));
+    set("middleman.detach-utility-actors", false);
+    set("middleman.enable-automatic-connections", true);
+    set("middleman.enable-tcp", true);
+    set("middleman.enable-udp", false);
+  }
 };
 
-behavior testee(stateful_actor<testee_state>* self) {
+// Used for the tests with the default multiplexer backend.
+class simple_config : public actor_system_config {
+public:
+  simple_config() {
+    load<caf::io::middleman>();
+    set("middleman.enable-automatic-connections", true);
+    set("middleman.enable-tcp", true);
+    set("middleman.enable-udp", false);
+  }
+};
+
+class fixture {
```

Comment:
Is there a particular reason not to use the `belt_fixture` form the `io_dsl.hpp` here?